Client/Frontend/Brave Notifications/BraveNotification.swift

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+public struct BraveNotificationPriority: Comparable {
+  var value: Int
+  static let low: Self = .init(value: 0)
+  static let high: Self = .init(value: 1)
+  
+  public static func < (lhs: BraveNotificationPriority, rhs: BraveNotificationPriority) -> Bool {
+    lhs.value < rhs.value
+  }
+}
+
+public enum DismissPolicy {
+  case automatic(after: TimeInterval = 30)
+  case explicit
+}
+
+public protocol BraveNotification: AnyObject {
+  var priority: BraveNotificationPriority { get }
+  var view: UIView { get }
+  var dismissAction: (() -> Void)? { get set }
+  var dismissPolicy: DismissPolicy { get }
+  var id: String { get }
+  var isHorizontalSwipe: Bool { get }
```

Comment:
What is `isHorizontalSwipe` for? This seems specific to rewards notifications

Reply:
It is technically for rewards notificaitons. but this value is needed in `BraveNotificationPresenter` to reset the timer when user gives a vertical swipe when notification itself is in dislike mode (isHorizontalSwipe == true).